### PR TITLE
Remove guide mutation events

### PR DESCRIPTION
**Pull request by hamishwillee**
This removes [Guide/Events/Mutation_events](https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Mutation_events) as part of fixing #2245.

In summary, the guide topic is almost identical to the webapi page [MutationEvent](https://developer.mozilla.org/en-US/docs/Web/API/MutationEvent). So this mostly just deletes the guide page, updates links to `MutationEvent`, and adds a redirect to `MutationEvent`

### CODE REVIEW

File: files/en-us/web/guide/events/index.html [hunk 1]
```diff
@@ -33,7 +33,8 @@
 	<li>both <a href="/en-US/docs/Web/API/Touch_events">touch events</a> and the earlier <a href="/en-US/docs/Web/API/Touch_events/Mozilla_experimental_events">Mozilla experimental touch events</a>, now deprecated/removed.</li>
 </ul>
 
-<p>The <strong>modification of the web page</strong> in structure or content might trigger some events, as explained in the <a href="/en-US/docs/Web/Guide/Events/Mutation_events">mutation events page</a>, but the use of these events has been deprecated in favor of the lighter <a href="/en-US/docs/Web/API/MutationObserver">Mutation Observer</a> approach.</p>
+<p>The <strong>modification of the web page</strong> in structure or content might trigger some events, as explained in the <a href="/en-US/docs/Web/API/MutationEvent">Web APIs
+MutationEvent</a>, but the use of these events has been deprecated in favor of the lighter <a href="/en-US/docs/Web/API/MutationObserver">Mutation Observer</a> approach.</p>
```
Comment (chrisdavidmills) [inline files/en-us/web/guide/events/index.html:37]: "WebAPIs MutationEvent" seems like a strange/awkward choice for link text. Wouldn't something like "MutationEvent reference page" be better?

File: files/en-us/web/api/mutationevent/index.html [hunk 2]
```diff
@@ -11,19 +11,21 @@
...
 
-<p><a href="/en-US/docs/Web/API/MutationObserver">Mutation Observers</a> are the proposed replacement for mutation events in DOM4. They are expected to be included in Firefox 14 and <a href="http://updates.html5rocks.com/2012/02/Detect-DOM-changes-with-Mutation-Observers">Chrome 18</a>.</p>
+<p><a href="/en-US/docs/Web/API/MutationObserver">Mutation Observers</a> have replaced mutation events in DOM4. They have been supported in <a href="/en-US/docs/Web/API/MutationObserver#browser_compatibility">most popular browsers for some years</a>.</p>
 
-<p>The practical reasons to avoid the mutation events are <strong>performance issues</strong> and <strong>cross-browser support</strong>.</p>
+<p>In addition, they should be avoided because they have <strong>performance issues</strong> and <strong>poor cross-browser support</strong> (as described in the following sections).</p>
```
Comment (chrisdavidmills) [inline files/en-us/web/api/mutationevent/index.html:28]: I would replace "they" wth "mutation events" in this sentence. Your structural update is much better, but I don't think it is clear enough what "they" is referring to from the previous sentence.

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Just a couple of teensy commments (just one waaaaafer-thin mint...)

File: files/en-us/web/guide/events/index.html (hunk 1 above)
Comment (hamishwillee) [reply files/en-us/web/guide/events/index.html:37]: Absolutely. FYI this page will eventually be deleted.

File: files/en-us/web/events/index.html [hunk 3]
```diff
@@ -81,6 +81,19 @@ <h2 id="Event_index">Event index</h2>
 		<td>Events fired on <a href="/en-US/docs/Web/API/IDBDatabase#events"><code>IDBDatabase</code></a>, <a href="/en-US/docs/Web/API/IDBOpenDBRequest#events"><code>IDBOpenDBRequest</code></a>, <a href="/en-US/docs/Web/API/IDBRequest#events"><code>IDBRequest</code></a>, <a href="/en-US/docs/Web/API/IDBTransaction#events"><code>IDBTransaction</code></a>.</td>
 	</tr>
 
+  <tr>
```
Comment (hamishwillee) [inline files/en-us/web/events/index.html:84]: @chrisdavidmills Thanks for the review. I fixed up as per your suggestions. This change is new - as previously discussed, addition of mutation events to the event reference with the sole purpose of pointing out that we now use mutation observers for this now.

**Comment by hamishwillee** [comment]
@chrisdavidmills PS Do you think it is "worth" adding BCD for MutationEvent?  Essentially this would show that the Mutation event is more than just deprecated - but actually removed in most places.

My leaning is "not worth it", but you might feel differently.

**Review by chrisdavidmills — APPROVED** [review]
I think this is looking good now; thanks @hamishwillee !

I'd agree that `MutationEvent` BCD is not really worth it.

**Comment by hamishwillee** [comment]
@chrisdavidmills 

> I'd agree that MutationEvent BCD is not really worth it.

Except if we decide that BCD is the canonical source of deprecation info :-0

**Pull request merged**